### PR TITLE
SurveyMonkey: added OAuth for Eu and Ca

### PR DESCRIPTION
**Pull request by darynaishchenko**
## What
*Describe what the change is solving*
*It helps to add screenshots if it affects the frontend.*
SurveyMonkey: added OAuth for Eu and Ca
added api url change for different origins of SurveyMonkey Accaunt datacenter 

## How
*Describe the solution*

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

**Review by arsenlosenko — APPROVED** [review]

**Review by bazarnov — APPROVED** [review]

File: airbyte-oauth/src/main/java/io/airbyte/oauth/flows/SurveymonkeyOAuthFlow.java
Comment (girarda) [inline airbyte-oauth/src/main/java/io/airbyte/oauth/flows/SurveymonkeyOAuthFlow.java:56]: this should fail if "origin" != "USA" (eg if we add an enum value and forget to update this block)

File: airbyte-oauth/src/main/java/io/airbyte/oauth/flows/SurveymonkeyOAuthFlow.java
Comment (girarda) [inline airbyte-oauth/src/main/java/io/airbyte/oauth/flows/SurveymonkeyOAuthFlow.java:79]: is there a reason not to reuse the logic in https://github.com/airbytehq/airbyte/pull/17964/files#diff-5c30eabeb5bb52b1031725d5bc944852ffbfaa1b71491aa249978109a38eee64R48 ?

File: airbyte-oauth/src/main/java/io/airbyte/oauth/flows/SurveymonkeyOAuthFlow.java
Comment (darynaishchenko) [reply airbyte-oauth/src/main/java/io/airbyte/oauth/flows/SurveymonkeyOAuthFlow.java:56]: the reason we added origin is a different urls for authorization, USA is default, but if someone wants to add new origin he should know about it, because it's origin purpose. I can add comment in spec.json about it.

File: airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/spec.json
Comment (grubberr) [inline airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/spec.json:7]: Do we need access_token here?

**Review by grubberr — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/spec.json
Comment (grubberr) [inline airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/spec.json:27]: Do we need oneOf here?

**Review by grubberr — CHANGES_REQUESTED** [review]

**Comment by grubberr** [comment]
what about ?

```python
@staticmethod
    def get_authenticator(config: Mapping[str, Any]):
        token = config["credentials"]["access_token"]
        if not token:
            token = config["access_token"]
        return TokenAuthenticator(token=token)
```

**Comment by grubberr** [comment]
@darynaishchenko 

you move instance-wide params "client_id", "secret_id" to "credentials"
is it ok? Did you tested oAuth with such migration?

**Review by grubberr — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/spec.json
Comment (grubberr) [inline airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/spec.json:67]: Why order: "1" ?

**Review by grubberr — CHANGES_REQUESTED** [review]

File: airbyte-config/init/src/main/resources/seed/source_definitions.yaml
Comment (grubberr) [inline airbyte-config/init/src/main/resources/seed/source_definitions.yaml:1058]: you don't need to bump here

**Review by grubberr — CHANGES_REQUESTED** [review]

File: airbyte-config/init/src/main/resources/seed/source_specs.yaml
Comment (grubberr) [inline airbyte-config/init/src/main/resources/seed/source_specs.yaml:11096]: you don't need to bump here

File: airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/spec.json
Comment (grubberr) [inline airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/spec.json:48]: order: 3

**Review by grubberr — CHANGES_REQUESTED** [review]

**Comment by alafanechere** [comment]
@darynaishchenko could you please implement the requested changes from @grubberr . This is blocking the resolution of an OC issue. Let me know if you're blocked

**Comment by darynaishchenko** [comment]
> @darynaishchenko could you please implement the requested changes from @grubberr . This is blocking the resolution of an OC issue. Let me know if you're blocked

implemented, waiting for @grubberr approve

File: airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/spec.json
Comment (alafanechere) [inline airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/spec.json:7]: Isn't `credentials` required field?

File: airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/spec.json
Comment (alafanechere) [inline airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/spec.json:26]: Aren't `client_secret` and `client_id` required?

File: airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/spec.json
Comment (darynaishchenko) [reply airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/spec.json:7]: we don't need to add `credentials`, `client_secret` and `client_id` to required fields, because it will break connection for old users.

File: airbyte-oauth/src/main/java/io/airbyte/oauth/flows/SurveymonkeyOAuthFlow.java
Comment (girarda) [inline airbyte-oauth/src/main/java/io/airbyte/oauth/flows/SurveymonkeyOAuthFlow.java:59]: the error message should include the unknown origin to help the user troubleshoot the issue

File: airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/source.py
Comment (girarda) [inline airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/source.py:63]: can we add a unit test for this change?

File: airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/spec.json
Comment (girarda) [inline airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/spec.json:79]: why are all of these new fields needed in the spec?

File: airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/spec.json
Comment (alafanechere) [reply airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/spec.json:7]: You're right but I think we need to make an arbitrage there: 
1. Break the old configurations but have a clear specification
2. Do not require this field, which can lead to multiple configuration errors we'll have to manage downstream.
@YowanR can you please suggest what to do in this scenario?

File: airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/source.py
Comment (grubberr) [reply airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/source.py:64]: I mean if not "credentials" section "access_token" has to be in object root

**Review by grubberr — APPROVED** [review]

File: airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/source.py
Comment (darynaishchenko) [reply airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/source.py:63]: done

File: airbyte-oauth/src/main/java/io/airbyte/oauth/flows/SurveymonkeyOAuthFlow.java
Comment (darynaishchenko) [reply airbyte-oauth/src/main/java/io/airbyte/oauth/flows/SurveymonkeyOAuthFlow.java]: done

File: airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/spec.json
Comment (darynaishchenko) [reply airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/spec.json:79]: this fields are used for auth with `origin` field

**Review by girarda — APPROVED** [review]
approved pending a decision on whether we should break old configurations or manage potential errors downstream https://github.com/airbytehq/airbyte/pull/17964/files#r1015665539

File: airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/spec.json
Comment (alafanechere) [reply airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/spec.json:7]: Option B will allow users not to fill the `credentials` field, leading to a failed connection check. 
This will produce on-call Sentry errors. 
A workaround would be to add a new logic to the `check`: Make the check return `false` if these value are empty. This will handle this case gracefully and not create OC issues. Wdyt @darynaishchenko / @girarda ?

Comment (darynaishchenko) [reply airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/spec.json:7]: @alafanechere for me looks good, I can add this logic to the `check`

Comment (girarda) [reply airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/spec.json:7]: I'm fine with option B.

@YowanR for context: the downside of Option is B is we need to maintain logic for every breaking change. The idea situation would be to support [config migration](https://github.com/airbytehq/airbyte/issues/5396)

Comment (YowanR) [reply airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/spec.json:7]: There was an update to the config migration EPIC. Is that sufficient to support the better option now?

Comment (darynaishchenko) [reply airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/spec.json:7]: @alafanechere added credential check, please take a look

**Review by alafanechere — APPROVED** [review]

**Review by lazebnyi — APPROVED** [review]

**Pull request merged**
